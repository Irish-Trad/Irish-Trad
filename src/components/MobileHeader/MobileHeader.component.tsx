import React, { useState } from 'react'
import Navbar from 'components/Navbar'

import {
  HeaderContainer,
  Logo,
  InternalLink,
  StyledIcon,
} from './MobileHeader.styles'

const MobileHeader: React.FC = () => {
  const [hidden, toggleHidden] = useState(false)

  return (
    <HeaderContainer>
      <Logo>
        <InternalLink to="/" onClick={() => toggleHidden(false)}>
          Irish-Trad.net
        </InternalLink>
        <StyledIcon onClick={() => toggleHidden(!hidden)} icon="bars" />
        {hidden ? <Navbar toggleHidden={toggleHidden} /> : null}
      </Logo>
    </HeaderContainer>
  )
}

export default MobileHeader
