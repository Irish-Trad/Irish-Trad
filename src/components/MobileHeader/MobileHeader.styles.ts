import styled from 'styled-components'

import { Link } from 'react-router-dom'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export const HeaderContainer = styled.header`
  background: linear-gradient(to bottom, green, darkgreen);
  color: white;
  font-family: meathflfregular;
  font-size: 24px;

  @media screen and (min-width: 760px) {
    display: none;
  }
`

export const Logo = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  align-items: center;
`

export const InternalLink = styled(Link)`
  text-decoration: none;
  color: white;
  padding: 12px 24px;

  &:hover {
    color: black;
    background-color: white;
  }
`

export const StyledIcon = styled(FontAwesomeIcon)`
  font-size: 24px;
  color: white;
  padding: 13px 24px;
  cursor: pointer;

  &:hover {
    color: black;
    background-color: white;
  }

  @media screen and (min-width: 850px) {
    display: none;
  }
`
