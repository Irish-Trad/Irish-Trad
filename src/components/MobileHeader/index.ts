export { default } from './MobileHeader.component'
