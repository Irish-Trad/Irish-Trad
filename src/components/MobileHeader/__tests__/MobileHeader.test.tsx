import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { HashRouter } from 'react-router-dom'
import MobileHeader from '../MobileHeader.component'

describe('MobileHeader', () => {
  afterEach(() => {
    cleanup()
  })

  it('Snapshot', () => {
    const snapshot = render(
      <HashRouter>
        <MobileHeader />
      </HashRouter>
    )

    expect(snapshot).toMatchSnapshot()
  })

  it('MobileHeader should render all possible tabs', () => {
    const { getByText } = render(
      <HashRouter>
        <MobileHeader />
      </HashRouter>
    )

    expect(getByText('Irish-Trad.net')).toBeInTheDocument()
    expect(getByText('Tablature')).toBeInTheDocument()
    expect(getByText('Lessons')).toBeInTheDocument()
    expect(getByText('Chords')).toBeInTheDocument()
    expect(getByText('Clef to TAB')).toBeInTheDocument()
    expect(getByText('Terminology')).toBeInTheDocument()
    expect(getByText('Community')).toBeInTheDocument()
    expect(getByText('Newsletter')).toBeInTheDocument()
  })
})
