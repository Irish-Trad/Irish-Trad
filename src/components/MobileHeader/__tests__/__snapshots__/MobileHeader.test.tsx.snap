// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileHeader Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-AxhUy jcanCe"
      >
        <div
          class="sc-AxgMl dFREyc"
        >
          <a
            class="sc-AxheI hPJXOh"
            href="#/"
          >
            Irish-Trad.net
          </a>
          <i
            class="fa"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-AxhUy jcanCe"
    >
      <div
        class="sc-AxgMl dFREyc"
      >
        <a
          class="sc-AxheI hPJXOh"
          href="#/"
        >
          Irish-Trad.net
        </a>
        <i
          class="fa"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
